### PR TITLE
Fixes validity errors while editing shifts and indentation errors in the file

### LINKED ISSUE #228: 404 error when editing shifts
Hi! Ran into this after a fresh pull. After editing a shift, when we click on the save button, an error page pops up

k2

### PR DESCRIPTION
**Pull request by smarshy**
Fixes #228 and #236 

The previous code after a recent merge seemed to have some indentation issues and duplicated code :worried: .due to which the above error were faced

After these changes:
1. It is checked that the shift dates lie between the job dates while editing
2. The concerns in #195 for logging hours are addressed
3. Merged with latest change to check time for edit shifts
4. Added time check for create shifts

Please do crosscheck and suggest changes

### CODE REVIEW

**Comment by tapasweni-pathak** [comment]
This now has conflicts, can you please see and fix it.

**Comment by smarshy** [comment]
@tapasweni-pathak Sure :)

**Comment by smarshy** [comment]
@tapasweni-pathak Done  :thumbsup:  Additional commit to add time check for create shift has been added. Please review :)

File: vms/shift/views.py [hunk 1]
```diff
@@ -82,23 +79,23 @@ def add_hours_manager(request, shift_id, volunteer_id):
...
+                        return HttpResponseRedirect(reverse(
+                            'shift:manage_volunteer_shifts',
+                            args=(volunteer_id, )
+                            ))
+                    else:
+                        messages.add_message(request, messages.INFO, 'End time should be greater than start time')
+                        return render(
+                            request,
+                            'shift/add_hours_manager.html',
+                            {'form': form, 'shift_id': shift_id, 'volunteer_id': volunteer_id, }
```
Comment (tapasweni-pathak) [inline vms/shift/views.py]: @smarshy I don't think you have changed anything here. Still the diff is coming. Is it due to spacing issues?

Comment (smarshy) [inline vms/shift/views.py]: @ Yes, in the second [ commit](https://github.com/smarshy/vms/commit/16fb2ead0eb4d05ff208a67dd88f6800451d323a)  I have just fixed the indentation for edit hours, add hours, add hours manager, edit hours manager. The code seemed to work fine and no issues were reported.

**Comment by tapasweni-pathak** [comment]
Thank you. :) :+1:

**Pull request merged**